openai/openai.go

```diff
@@ -175,18 +176,29 @@ func New(opts ...Option) (*Client, error) {
 	if cfg.proxyURL != "" {
 		proxy, _ := url.Parse(cfg.proxyURL)
 		httpClient.Transport = &http.Transport{
-			Proxy: http.ProxyURL(proxy),
+			Proxy:               http.ProxyURL(proxy),
+			MaxIdleConns:        100,
+			MaxIdleConnsPerHost: 100,
+			TLSHandshakeTimeout: 5 * time.Second,
 		}
 	} else if cfg.socksURL != "" {
 		dialer, err := proxy.SOCKS5("tcp", cfg.socksURL, nil, proxy.Direct)
 		if err != nil {
 			return nil, fmt.Errorf("can't connect to the proxy: %s", err)
 		}
 		httpClient.Transport = &http.Transport{
-			Dial: dialer.Dial,
+			Dial:                dialer.Dial,
+			MaxIdleConns:        100,
+			MaxIdleConnsPerHost: 100,
+			TLSHandshakeTimeout: 5 * time.Second,
+		}
+	} else {
+		httpClient.Transport = &http.Transport{
+			MaxIdleConns:        100,
+			MaxIdleConnsPerHost: 100,
+			TLSHandshakeTimeout: 5 * time.Second,
 		}
 	}
-
 	c.HTTPClient = httpClient
 	instance.client = openai.NewClientWithConfig(c)
```

Comment:
Overall, the code patch looks good and follows best practices. Here are a few suggestions for improvement:

1. Add error handling: There are a few places in the code where errors are not being handled properly. For example, if the proxy URL is malformed, the code will continue running without returning an error. It's important to handle errors properly to prevent unexpected behavior.

2. Remove unnecessary code: The line `import "time"` is not being used, so it can be safely removed.

3. Use constants for magic numbers: The values `100` and `5` are used multiple times in the code. It's a good practice to define constants for these values to make the code more readable and easier to maintain.

4. Use a logger: Instead of printing errors and debug information to the console, consider using a logger to write to a file or a log aggregation service. This makes it easier to track down issues and monitor the application.

5. Consider using a connection pool: The `http.Transport` object is being created with a maximum number of idle connections and max idle connections per host. However, there is no connection pool being used to manage these connections. Consider using a connection pool library like `github.com/fatih/pool` to manage connections more efficiently.

6. Use a context: The `http.Client` object is being created without a context. Consider using a context to manage the lifecycle of the client and cancel requests when necessary.